mobile/library/common/engine.cc

```diff
@@ -184,9 +186,16 @@ envoy_status_t Engine::recordCounterInc(const std::string& elements, envoy_stats
   ASSERT(dispatcher_->isThreadSafe(), "pulse calls must run from dispatcher's context");
   Stats::StatNameTagVector tags_vctr =
       Stats::Utility::transformToStatNameTagVector(tags, stat_name_set_);
-  std::string name = Stats::Utility::sanitizeStatsName(elements);
-  Stats::Utility::counterFromElements(*client_scope_, {Stats::DynamicName(name)}, tags_vctr)
-      .add(count);
+  if (sink_sanitization_enabled_) {
+    std::string buffer;
+    absl::string_view name = Stats::Utility::sanitizeStatsName(elements, buffer);
```

Comment:
It's confusing why sanitizeStatsName is being called in both branches of the if-statement. They are two different signatures. Should they they have two different names?

Reply:
I wanted to avoid renaming the common metric name sanitizer sanitizeStatsName() as it is the right sink agnostic name and sanitizes all metric names irrespective of sinks they will be sent to like stripping of leading/trailing '.' and null char only. 
I though leveraging polymorphism will keep the intent simpler - one that doesn't take buffer will eventually be deprecated in favour of one that takes buffer to avoid string copy when possible. It will be easy to remember and deprecate after few months.